File: inflearn/1/hello-spring/hello-spring/src/main/java/com/example/hellospring/controller/HelloController.java
package com.example.hellospring.controller;

import lombok.Data;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
public class HelloController {

    @GetMapping("hello")
    public String hello(Model model) {
        model.addAttribute("data", "홍길동!!");  // key=data : value=홍길동!!
        return "hello";  // resources.templates.hello.html 로 연결
    }

    @GetMapping("hello-mvc")
    public String helloMvc(@RequestParam(value = "name", required = true) String name, Model model) {
        model.addAttribute("name", name);
        return "hello-template";
    }

    @GetMapping("hello-string")
    @ResponseBody
    public String helloString(@RequestParam(value = "name", required = true) String name) {
        return "hello string " + name;
    }

    @GetMapping("hello-api")
    @ResponseBody
    public Hello helloApi(@RequestParam(value = "name", required = true) String name) {
        Hello hello = new Hello();
        hello.setName(name);
        return hello;
    }

    @Data
    private static class Hello {
        private String name;
    }
}

File: kotlin_hello_world/src/main/kotlin/com/example/kotlinhelloworld/HelloController.kt
package com.example.kotlinhelloworld

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("")
class HelloController {
    @GetMapping("/hello")
    fun hello(@RequestParam("name", required = false, defaultValue = "World") name: String) = "Hello, $name!"

    @GetMapping("/")
    fun getList():List<Message> {
        return listOf(
            Message("2", "hello"),
            Message("2", "good bye"),
            Message("3", "sure"),
        );
    }
}
